demos/checkbox.html

```diff
@@ -14,50 +14,145 @@
+        background: #262626;
+        color: white;
+      }
+
+      .dark-theme > h2 {
+        margin-top: 0;
+      }
+
+      /* Example of manually theming a checkbox. Note that this does not take JS-provided animations
+         into account. */
+      .dark-theme .mdl-checkbox__native-control:checked:not(:disabled) ~ .mdl-checkbox__background,
+      .dark-theme .mdl-checkbox__native-control:indeterminate:not(:disabled) ~ .mdl-checkbox__background {
+        background-color: #E91E63;
+        border-color: #E91E63;
+      }
+      .dark-theme .mdl-checkbox__background::before {
+        background-color: #E91E63;
+      }
+      .dark-theme .mdl-checkbox__checkmark__path {
+        stroke: #262626 !important;
+      }
+      .dark-theme .mdl-checkbox__mixedmark {
+        background-color: #262626;
+      }
+    </style>
   </head>
   <body>
     <main>
-      <h1>MDL Checkbox Hello</h1>
+      <h1>MDL Checkbox</h1>
       <section>
-        <h2>Auto-initialized checkbox</h2>
-        <!-- TODO change from "md-" to "mdl-" -->
-        <div class="md-checkbox-wrapper">
-          <div class="md-checkbox-wrapper__layout">
-            <div class="md-checkbox" data-mdl-auto-init="MDLCheckbox">
+        <h2>Basic Example, no Javascript</h2>
+        <div class="mdl-checkbox-wrapper">
+          <div class="mdl-checkbox-wrapper__layout">
+            <div class="mdl-checkbox">
+              <input type="checkbox"
+                     id="basic-checkbox"
+                     class="mdl-checkbox__native-control"
+                     aria-labelledby="basic-checkbox-label"/>
+              <div class="mdl-checkbox__background">
+                <svg version="1.1"
+                     class="mdl-checkbox__checkmark"
+                     xmlns="http://www.w3.org/2000/svg"
+                     viewBox="0 0 24 24"
+                     xml:space="preserve">
+                  <path class="mdl-checkbox__checkmark__path"
+                        fill="none"
+                        stroke="white"
+                        d="M1.73,12.91 8.1,19.28 22.79,4.59"/>
+                </svg>
+                <div class="mdl-checkbox__mixedmark"></div>
+              </div>
+            </div>
+            <label for="basic-checkbox" id="basic-checkbox-label">This is my checkbox</label>
+          </div>
+        </div>
+        <button type="button" onClick="document.getElementById('basic-checkbox').indeterminate = true;">Make indeterminate</button>
+        <button type="button" onClick="this.parentElement.hasAttribute('dir') ? this.parentElement.removeAttribute('dir') : this.parentElement.setAttribute('dir', 'rtl');">Toggle RTL</button>
+        <button type="button" onClick="document.querySelector('.mdl-checkbox-wrapper').classList.toggle('mdl-checkbox-wrapper--align-end');">Toggle Align End</button>
+        <input type="checkbox" id="disable-element-basic" onchange="document.getElementById('basic-checkbox').disabled = this.checked;">
```

Comment:
It is a bit inceptiony. But, it also shows in an easy demo how a developer can handle events in an actual application of the component.

In lieu of a checkbox we could just have a "Toggle Disabled" button which does `e => e.target.disabled = !e.target.disabled;` on click. That way all the toggles are buttons vs this one checkbox node.